src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformContentViewGroup.java

```diff
@@ -28,13 +28,12 @@ public PlatformContentViewGroup(Context context, AttributeSet attrs, int defStyl
 
     /**
      * Set by C#, determining if we need to call getClipPath()
+     * Intentionally invalidates the view in case clip changed
      * @param hasClip
      */
     protected final void setHasClip(boolean hasClip) {
-        if (this.hasClip != hasClip) {
```

Comment:
Why do we remove the `if`? If the value for `hasClip` hasn't changed, why does it need to run `invalidate()`?

Does it actually need to be called in a different place in the stack? In .NET MAUI's C# layer, perhaps it would _know_ what changed?

Reply:
@jonathanpeppers I've just done what you asked: revert your PR and use `invalidate` instead of `postInvalidate`.

That said:
- setting a boolean field value has very likely no impact on performance
- this method is being called from the C# layer when the `Clip` changes, that's why we always need to invalidate
  - calling it from C# would cause two JNI roundtrips for nothing

Reply:
> setting a boolean field value has very likely no impact on performance

Setting the field isn't the question, but the fact that `invalidate()` is called. Why do we need call `invalidate()` if `true` is set to `true`? or `false` set to `false`?

Reply:
Because the clip path or the shadow details like color, radius,... have changed.

Reply:
we should call these methods something like `invalidateShadow(bool willDrawShadow)`

Reply:
If the clip path or shadow changes, should those be the things that call `invalidate()`?

Reply:
The reason to not move `invalidate` call there is just to avoid two round trips on JNI. So this is just a naming problem in my opinion.